Aesthetic changes to criteria.md

MaxGhenis opened the pull request:
See commit descriptions

MaxGhenis commented on Catalog/library_criteria.md:
```diff
@@ -1,46 +1,43 @@
 1. Projects MUST have installation directions.
 1. Project MUST be mirrored in the same GitHub organization as PSL, and therefore they MUST be under version control.
-1. Projects MUST use a consistent versioning scheme, which SHOULD be [semantic versioning](https://semver.org/). If projects want to use the PSL Package-Builder tool to distribute packages via the Anaconda Cloud PSLmodels channel, there are [additional MUST criteria](https://github.com/PSLmodels/Package-Builder#using-package-builders-pbrelease-tool).
+1. Projects MUST use a consistent versioning scheme. If projects want to use the PSL Package-Builder tool to distribute packages via the Anaconda Cloud PSLmodels channel, there are [additional MUST criteria](https://github.com/PSLmodels/Package-Builder#using-package-builders-pbrelease-tool).
```
Is Package-Builder still relevant?

jdebacker replied:
I do not think it is worth mentioning Package-Builder here. cc @MattHJensen

MaxGhenis reviewed (commented):
.

jdebacker commented on Catalog/library_criteria.md:
```diff
@@ -56,10 +50,4 @@ Community Criteria
-Interoperability Criteria
---------------------------
-
 1. The source code SHOULD be written in an open source language.
```
Should this "Should" be moved up above the "may" lines?

jdebacker commented:
@MaxGhenis Thanks for this over due clean up!  If others agree, I like removing the package builder reference.  I have one suggestion on ordering in the lists of shoulds and mays, but don't feel strongly about it if you like the order as it is currently.

MaxGhenis commented:
@jdebacker Thanks for the suggestion, I moved that element up with other `SHOULD`s, and changed the verbiage to replace the repeated `MUST`/`SHOULD`/`MAY` with sections for required, recommended, and optional practices. I also removed Package-Builder.

Peter-Metz commented:
Thanks a lot @MaxGhenis and @jdebacker. This looks excellent to me. I'll merge this PR tomorrow if no one has other feedback. These changes will automatically be reflected on the website thanks to a GH action.
